backend/daemon/daemon_e2e_test.go

```diff
@@ -421,6 +421,84 @@ func TestSite(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestTrustedChanges(t *testing.T) {
```

Comment:
Added this tests @burdiyan It passes it please let me know if you want any other thing to check.

Reply:
I don't quite follow the "last trusted idx" trick in this implementation. Can you explain if this algorithm would handle the following situation properly?

Imagine we have the following DAG of changes, and we only trust Alice here. I don't understand if this algorithm will correctly return `E@Alice` as the latest trusted version.

![d2](https://github.com/mintterteam/mintter/assets/1793789/d8ad853e-2c16-4685-ae40-699a4eb54986)

Reply:
I think `entity.ApplyChange(change.chID, change.ch)` for chage `C` would not have any effect on the resulting entity if I stop at `E`, right? As long as I don't include `F` (that depends on `C` I should be fine. Nevertheless, how can I test a complex escenario like this? is there any test where we modify an entity in such a way? @burdiyan ?

Reply:
Something similar could be found in `backend/hyper/entity_test.go`. This is where we have more isolated tests for bare Entities, without depending on anything else. I guess you could manage to create a scenario similar to the picture above using these examples.

Reply:
I just realized that there might be even easier algorithm than the one I was proposing initially. And it's similar to how `ChangesGetPublicHeadsJSON` works.

1. Get all the changes for an entity from trusted editors.
2. For those changes, get their parents, and deduplicate them (e.g. use `DISTINCT` clause in SQLite).
3. Get the difference between IDs from step 1 and step 2. Those heads from your trusted peers.
4. Pass those heads to our common load entity function.

This algorithm should correctly handle the situation when a trusted change depends on non-trusted changes, like the example picture from above. And it can be done within a single SQLite query, similar to `ChangesGetPublicHeadsJSON`.

Reply:
I'am confused with step 1, if only trusted changed are taken into consideration, wouldn't change `B`and `C` be discarded by Alice At the time of loading the entity before apply change `E`, @burdiyan ?

Reply:
Yes, that's how you discard untrusted changes, that's what we want. But we need to find which of these changes are heads. Although they might not technically be heads, like in this example: the last trusted change has children, untrusted though. But we need to find from this selected set of changes which ones are heads. So for each selected change from step 1, we find their parents, and then we find which changes from step 1 are not parents of any other change, so we find heads. Then we walk back from heads and thus discover the full chain (which might contain untrusted changes).

I know it's confusing as hell, but I hope this help a bit.

Reply:
I finally managed to do it in sql. WITH RECURSIVE took me a while to understand but it's pretty easy actually once understood. You can have a final look in case I oversaw something @burdiyan

Reply:
I'm trying to understand this query, but I can't yet :( The nested CTE confuses me. Trying harder :)

Reply:
I'm closing this, and will open a separate comment on the query itself.